dubbo-metrics/dubbo-metrics-api/src/main/java/org/apache/dubbo/metrics/filter/observation/AbstractDefaultDubboObservationConvention.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.metrics.filter.observation;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.common.lang.Nullable;
+import org.apache.dubbo.common.utils.StringUtils;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.RpcContextAttachment;
+
+import static org.apache.dubbo.metrics.filter.observation.DubboObservation.LowCardinalityKeyNames.NET_PEER_NAME;
+import static org.apache.dubbo.metrics.filter.observation.DubboObservation.LowCardinalityKeyNames.NET_PEER_PORT;
+import static org.apache.dubbo.metrics.filter.observation.DubboObservation.LowCardinalityKeyNames.RPC_METHOD;
+import static org.apache.dubbo.metrics.filter.observation.DubboObservation.LowCardinalityKeyNames.RPC_SERVICE;
+import static org.apache.dubbo.metrics.filter.observation.DubboObservation.LowCardinalityKeyNames.RPC_SYSTEM;
+
+class AbstractDefaultDubboObservationConvention {
+    KeyValues getLowCardinalityKeyValues(Invocation invocation, RpcContextAttachment rpcContextAttachment) {
+        KeyValues keyValues = KeyValues.of(RPC_SYSTEM.withValue("apache_dubbo"));
+        String serviceName = StringUtils.hasText(invocation.getServiceName()) ? invocation.getServiceName() : readServiceName(invocation.getTargetServiceUniqueName());
+        keyValues = appendNonNull(keyValues, RPC_SERVICE, serviceName);
+        keyValues = appendNonNull(keyValues, RPC_METHOD, invocation.getMethodName());
+        keyValues = appendNonNull(keyValues, NET_PEER_NAME, rpcContextAttachment.getRemoteHostName());
+        if (rpcContextAttachment.getRemotePort() == 0) {
+            return keyValues;
+        }
+        int port = rpcContextAttachment.getRemotePort() != 0 ? rpcContextAttachment.getRemotePort() : rpcContextAttachment.getLocalPort();
```

Comment:
`NET_PEER_NAME` is design for provider side or consumer side?
If for provider side, use `org.apache.dubbo.rpc.RpcContext#getServiceContext.getRemoteAddress()` instead. `RpcContextAttachment` is design for attachment only. The `getRemotePort` here is for compatible purpose.
If for consumer side, it is not a good idea to track it because consumer can call for retry or boardcast. Thus, it is hard to fetch the actual remote address, but it is possible. :)

Reply:
Good catch, that's for the client https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-client

Reply:
I've added a commit that addresses most of the suggestions. The only problem is with this part here and I left it as it was cause I don't know how to retrieve this information in any other way

Reply:
> I've added a commit that addresses most of the suggestions. The only problem is with this part here and I left it as it was cause I don't know how to retrieve this information in any other way

Do you mean that you want to retrieve the remote port in consumer?

Reply:
I would like to retrieve the remote port and the remote host in the consumer.

Reply:
I have been submit a [PR](https://github.com/apache/dubbo/pull/11359) related this feature, which may helps you to retrieve the invoked remote invoker in `ClusterFilter#onResponse` and `ClusterFilter#onError`.

Reply:
I don't know if I did this correctly - if I didn't then I need someone to tell me how to do it properly.

Reply:
I have submitted the suggest changes below. 
In addition, I haven't figured out how to record this record if Dubbo's request occurs many times.